nuget/lib/dependabot/nuget/file_updater.rb

```diff
@@ -128,6 +133,13 @@ def original_declarations(dependency, requirement)
                "#{Regexp.escape(dependency.previous_version)}"/x
             ).to_s
           ]
+        elsif requirement.fetch(:file).casecmp(".config/dotnet-tools.json").zero?
+          [
+            dotnet_tools_json.content.match(
+              /"#{Regexp.escape(dependency.name)}"(?:.*):(?:.*){(?:.*)"version"(?:.*)
```

Comment:
I think this may be matching too much, because `.*` is greedy by default?

Like, in the example `dotnet-tools.json` file added as a test case, if both `botsay` and `dotnetsay` are locked at the same version, they would both get updates when updating the first dependency in the file (`botsay`)?

Reply:
The first double quotation would match one and both the other. E.g. "dotnetsay" : { "version" : "1.0.0" } would not conflict with "botsay" : { "version" : "1.0.0" }

As for the greedy match, i wouldn't get it to work with `\s` or `\S`. Do you have a suggestion?

Reply:
This is what I mean: 
https://rubular.com/r/xD4D1x1SlqDRkP. This regex would match both dependencies, and then when replacing I'm afraid it would update the versions of both. Does that make sense?

I would indeed try to use `\s`. Maybe something like

```suggestion
              /"#{Regexp.escape(dependency.name)}"(?:\s*):(?:\s*){(?:[^}]*)"version"(?:\s*):(?:\s*)
```

Also, can these be quoted using single quotes?

Reply:
Thanks for the insights here. I will make the RegEx change shortly. It seems to solve the conflict problem in the process? https://rubular.com/r/8owRHxVYfwXM4y

Reply:
If you can add an extra spec to cover the tricky case, that'd be awesome!

Reply:
Done.

Reply:
Thanks! Let me have a look!